contract_payment_mode/models/contract.py

```diff
@@ -26,7 +26,9 @@ def _prepare_invoice_data(self, contract):
             _prepare_invoice_data(contract)
         if contract.payment_mode_id:
             invoice_vals['payment_mode_id'] = contract.payment_mode_id.id
-            invoice_vals['partner_bank_id'] = \
-                contract.payment_mode_id.bank_id.id
+            invoice_vals['partner_bank_id'] = (
+                contract.partner_id.bank_ids and
```

Comment:
Use `contract.partner_id.bank_ids[:1].id or contract.payment_mode_id.bank_id.id)` for shorten the expression (and don't make another PR - I have told you how to do it in the other PR).